### ISSUE TITLE
Allow destination queues to re-execute the filter/transformer

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3454]
Regenerating the template (and connector properties) allows for some variable flexibility when queuing, but sometimes it's not enough. Sometimes you need to be able to some more complex logic on each queue attempt to reset certain destination settings. If users could choose to re-run the filter/transformer on each queue attempt, it would make queuing a lot more powerful/dynamic.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3558
Reporter: narupley
Created: 2014-12-19T17:01:42.000-0800

**Comment by rbeckman-nextgen** [comment]
Revision 7503: Added the ability for destination queues to re-execute the filter/transformer on each queue attempt. A new "Include Filter/Transformer" option was added to the advanced queue settings. If selected and the queue is set to Always, the filter/transformer will **not** be executed on the main message processing thread, but instead will always be executed in the queue thread.

Imported Comment. Original Details:
Author: narupley
Created: 2014-12-19T17:06:17.000-0800

**Comment by rbeckman-nextgen** [comment]
Verified that queued messages were re-run through filter/transformer rule/steps when this option is selected.

I created a simple transformer step to log out a variable taken from the global map that I populated in a global deploy script. I then created two separate channels both with channel writers as their destination connectors. The first channel that is responsible for routing to the second channel had queuing enabled with advanced settings to regenerate the template and include the filter/transformer. I stopped the downstream channel, and after allowing messages to queue up with several logger statements printed out, I changed the global script to put a different value in the global map, redeploying the second channel so that it was started and would reprocess messages. The logger statements that printed out were updated to what was now in the global map as opposed to what initially should have been there. I repeated this same test instead slightly changing my transformer's logger statements. These also reflected the new changes that were saved.

This process was repeated using a filter on the destination connector. I essentially allowed messages to queue up and then redeployed the channel with filtering criteria. Messages that otherwise should have been accepted were now filtered instead at this point.

I also tested the above, changing channel properties between each deploy. When messages were in the queue with the previous settings set to re-run the filter/transformer, after the channel was redeployed with the options removed, the filter/transformers were not rerun.

Imported Comment. Original Details:
Author: jaysenp
Created: 2015-01-20T17:48:58.000-0800
